ssstar/src/role_credentials_provider.rs

```diff
@@ -14,8 +14,22 @@ use std::{
 };
 use tokio::sync::Mutex;
 
+/// The configuration to assume bastion-based role
+#[derive(Debug)]
+struct BastionConfig {
+    /// Role-arn of bastion role
+    role_arn: String,
+    /// The name of the SSM external-ID parameter
+    external_id_name: String,
+    /// Session name for assuming the bastion-based role
+    session_name: String,
+}
```

Comment:
The concept of `bastion` role is private to `elastio` repository. It shouldn't be exposed in the `ssstar` crate. `ssstar` shouldn't even mention the name "bastion". Instead, it should accept the following parameters for the role to assume:
- ARN of the role
- Value of the external ID parameter (yes, the caller needs to resolve the external ID on its own); we need to modify our `asset-account-auth` crate in `elastio/elastio` repository to expose that information in the `ConnectorIdentity`
- Name of the session

Because `ssstar` is a generic tool this should be named just "role to assume", and not "bastion role to assume with eternal ID from this SSM parameter"; Convenience wrappers for working with the bastion role may be implemented in the `elastio/account`. Btw. the logic of assuming the `bastion` role may change in the future and it's better to reuse it. I think it's better to discuss this problem on a call. Please ping me when you have time for this

Reply:
I entirely removed this from ssstar, instead, i've added a new trait `CustomCredentialsUpdateCallback` which the user should implement and pass to ssstar

Reply:
@Veetaha I thought the `bastion` is something common to AWS because i've found a post https://docs.aws.amazon.com/managedservices/latest/userguide/using-bastions.html

but probably in our case it's something specific